### PR TITLE
Bug fixing, Authentication support , Time Aggregation and UI Improvement

### PR DESCRIPTION
**Pull request by Yasara123**
## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: components/org.wso2.carbon.status.dashboard.core/pom.xml [hunk 1]
```diff
@@ -111,6 +112,15 @@
...
             <artifactId>org.wso2.carbon.analytics.msf4j.interceptor.common</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.wso2.carbon.analytics-common</groupId>
+            <artifactId>org.wso2.carbon.database.query.manager</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.ocpsoft.prettytime</groupId>
+            <artifactId>prettytime</artifactId>
+            <version>4.0.1.Final</version>
```
Comment (suhothayan) [inline components/org.wso2.carbon.status.dashboard.core/pom.xml]: move the version to parent pom

File: components/org.wso2.carbon.status.dashboard.web/src/gadgets/chart-components/Legend/LegendItem.jsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright (c) WSO2 Inc. (http://wso2.com) All Rights Reserved.
```
Comment (suhothayan) [inline components/org.wso2.carbon.status.dashboard.web/src/gadgets/chart-components/Legend/LegendItem.jsx]: not the correct licence file

File: components/org.wso2.carbon.status.dashboard.web/src/gadgets/chart-components/Legend/LegendItem.jsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+export default class LegendItem extends Component {
+
+    constructor(props) {
+        super(props);
+        this._onMouseClick = this._onMouseClick.bind(this);
+        this._onMouseOver = this._onMouseOver.bind(this);
+        this.onMouseOver = this._onMouseOver.bind(this);
+    }
+
+
```
Comment (suhothayan) [inline components/org.wso2.carbon.status.dashboard.web/src/gadgets/chart-components/Legend/LegendItem.jsx]: remove spaces

File: components/org.wso2.carbon.status.dashboard.core/pom.xml (hunk 1 above)
Comment (Yasara123) [reply components/org.wso2.carbon.status.dashboard.core/pom.xml]: Fix 382f396

File: components/org.wso2.carbon.status.dashboard.web/src/gadgets/chart-components/Legend/LegendItem.jsx (hunk 2 above)
Comment (Yasara123) [reply components/org.wso2.carbon.status.dashboard.web/src/gadgets/chart-components/Legend/LegendItem.jsx]: Fix 4006cc1

File: components/org.wso2.carbon.status.dashboard.web/src/gadgets/chart-components/Legend/LegendItem.jsx (hunk 3 above)
Comment (Yasara123) [reply components/org.wso2.carbon.status.dashboard.web/src/gadgets/chart-components/Legend/LegendItem.jsx]: Fix 4006cc1

File: components/org.wso2.carbon.siddhi.metrics.core/src/main/java/org/wso2/carbon/siddhi/metrics/core/SiddhiBufferedEventsMetric.java [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+     * @param eventBufferHolder          Buffered object.
+     * @parambufferedEventsTrackerId An unique value to identify the  Buffered object.
+     */
+    @Override
+    public void registerEventBufferHolder(EventBufferHolder eventBufferHolder, String bufferedEventsTrackerId) {
+        if (registeredObjects.get(eventBufferHolder) == null) {
+            registeredObjects.put(eventBufferHolder, new ObjectMetric(eventBufferHolder, bufferedEventsTrackerId));
+            SiddhiMetricsManagement.getInstance().addComponent(siddhiAppName, bufferedEventsTrackerId);
+        }
+
```
Comment (gimantha) [inline components/org.wso2.carbon.siddhi.metrics.core/src/main/java/org/wso2/carbon/siddhi/metrics/core/SiddhiBufferedEventsMetric.java]: remove line

File: components/org.wso2.carbon.siddhi.metrics.core/src/main/java/org/wso2/carbon/siddhi/metrics/core/SiddhiBufferedEventsMetric.java [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+            this.name = name;
+            initMetric();
+        }
+
+        public String getName() {
+            return name;
+        }
+
+        private void initMetric() {
+                metricService.gauge(
```
Comment (gimantha) [inline components/org.wso2.carbon.siddhi.metrics.core/src/main/java/org/wso2/carbon/siddhi/metrics/core/SiddhiBufferedEventsMetric.java]: fix indentation

File: components/org.wso2.carbon.siddhi.metrics.core/src/main/java/org/wso2/carbon/siddhi/metrics/core/SiddhiLatencyMetric.java [hunk 6]
```diff
@@ -33,22 +33,21 @@
...
             if (isStatisticEnabled) {
-                SiddhiMetricsDataHolder.getInstance().getMetricManagementService().setMetricLevel(latencyTrackerId,
+                SiddhiMetricsDataHolder.getInstance().getMetricManagementService().setMetricLevel(SiddhiLatencyMetric.this.latencyTrackerId,
                         INFO);
             } else {
-                SiddhiMetricsDataHolder.getInstance().getMetricManagementService().setMetricLevel(latencyTrackerId,
+                SiddhiMetricsDataHolder.getInstance().getMetricManagementService().setMetricLevel(SiddhiLatencyMetric.this.latencyTrackerId,
                         OFF);
             }
                 return timer;
```
Comment (gimantha) [inline components/org.wso2.carbon.siddhi.metrics.core/src/main/java/org/wso2/carbon/siddhi/metrics/core/SiddhiLatencyMetric.java]: fix indentation

File: components/org.wso2.carbon.status.dashboard.core/src/main/java/org/wso2/carbon/status/dashboard/core/bean/SiddhiAppStatus.java [hunk 7]
```diff
@@ -52,7 +56,9 @@ public int getAge() {
     }
 
     public void setAge(int age) {
-        this.age = age;
+        //to make seconds
+        this.agetime = getTimeAgo(age);
+        this.age = age/1000;
```
Comment (gimantha) [inline components/org.wso2.carbon.status.dashboard.core/src/main/java/org/wso2/carbon/status/dashboard/core/bean/SiddhiAppStatus.java]: space after and before /

File: components/org.wso2.carbon.status.dashboard.core/src/main/java/org/wso2/carbon/status/dashboard/core/dbhandler/StatusDashboardMetricsDBHandler.java [hunk 8]
```diff
@@ -526,7 +545,7 @@ public List selectWorkerAggregatedThroughput(String workerId, long timeInterval,
      * @param query selection query.
      * @return the selected object.
      */
-    private List<List<Object>> selectGauge(String query) {
+    private List<List<Object>> selectGauge(String query,boolean isAggregated) {
```
Comment (gimantha) [inline components/org.wso2.carbon.status.dashboard.core/src/main/java/org/wso2/carbon/status/dashboard/core/dbhandler/StatusDashboardMetricsDBHandler.java]: space after comma

File: components/org.wso2.carbon.status.dashboard.web/src/gadgets/chart-components/Legend/LegendItem.jsx [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+    render() {
+        return (
+            <div
+                onClick={this.props.onClick}
+                onMouseOver={this.props.onMouseOver}
+                onMouseOut={this.props.onMouseOut}
+            >
+                <span
+
```
Comment (gimantha) [inline components/org.wso2.carbon.status.dashboard.web/src/gadgets/chart-components/Legend/LegendItem.jsx]: remove unnecessary spaces and lines and format

File: components/org.wso2.carbon.status.dashboard.web/src/gadgets/chart-components/NumberChart.jsx [hunk 10]
```diff
@@ -11,109 +11,102 @@
...
-                    style={{fontSize: 30}}
+                    style={{ fontSize: 30 }}
                 />
                 <VictoryLabel
-                    textAnchor="middle" verticalAnchor="middle"
-                    x={width / 2} y={height / 2}
-
-                    text={(value===null ? value : value.toFixed(3))}
-                    style={{fontSize: 45}}
+                    textAnchor="middle"
```
Comment (gimantha) [inline components/org.wso2.carbon.status.dashboard.web/src/gadgets/chart-components/NumberChart.jsx]: formatting issues.. spaces before and after the operator

Comment (gimantha) [reply components/org.wso2.carbon.status.dashboard.web/src/gadgets/chart-components/NumberChart.jsx]: Check other places as well

File: components/org.wso2.carbon.status.dashboard.web/src/gadgets/chart-components/PieChart.jsx [hunk 11]
```diff
@@ -95,64 +86,56 @@ export default class PieCharts extends React.Component {
...
+                                    fill: chartArray[0].colorScale[chartArray[0].colorIndex],
                                 };
                             }
-
                         } else {
                             dataSets[dataSetName] = {
                                 x: dataSetName,
                                 y: datum[xIndex],
-                                fill: chartArray[0].colorScale[chartArray[0].colorIndex]
+                                fill: chartArray[0].colorScale[chartArray[0].colorIndex],
```
Comment (gimantha) [inline components/org.wso2.carbon.status.dashboard.web/src/gadgets/chart-components/PieChart.jsx]: remove the last comma.. it is not necessary

File: components/org.wso2.carbon.status.dashboard.web/src/gadgets/chart-components/PieChart.jsx [hunk 12]
```diff
@@ -165,12 +148,10 @@ export default class PieCharts extends React.Component {
             orientation,
             legend,
             scatterPlotRange,
-            randomUpdater
+            randomUpdater,
```
Comment (gimantha) [inline components/org.wso2.carbon.status.dashboard.web/src/gadgets/chart-components/PieChart.jsx]: comma is not needed. please check

File: components/org.wso2.carbon.status.dashboard.web/src/gadgets/chart-components/PieChart.jsx [hunk 13]
```diff
@@ -221,28 +199,28 @@ export default class PieCharts extends React.Component {
                                     return [
                                         {
                                             target: 'data',
-                                            mutation: this._handleMouseEvent
-                                        }
+                                            mutation: this._handleMouseEvent,
```
Comment (gimantha) [inline components/org.wso2.carbon.status.dashboard.web/src/gadgets/chart-components/PieChart.jsx]: comma not needed..

Comment (gimantha) [reply components/org.wso2.carbon.status.dashboard.web/src/gadgets/chart-components/PieChart.jsx]: check other such places

File: components/org.wso2.carbon.stream.processor.statistics/src/main/java/org/wso2/carbon/stream/processor/statistics/service/SiddhiAppRuntimeServiceComponent.java [hunk 14]
```diff
@@ -0,0 +1,81 @@
...
+import org.osgi.service.component.annotations.ReferenceCardinality;
+import org.osgi.service.component.annotations.ReferencePolicy;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.wso2.carbon.config.ConfigurationException;
+import org.wso2.carbon.stream.processor.core.SiddhiAppRuntimeService;
+import org.wso2.carbon.stream.processor.statistics.internal.StreamProcessorStatisticDataHolder;
+
+/**
+ * This is OSGi-components to register config provider class.
```
Comment (gimantha) [inline components/org.wso2.carbon.stream.processor.statistics/src/main/java/org/wso2/carbon/stream/processor/statistics/service/SiddhiAppRuntimeServiceComponent.java]: Is this class level comment correct?

File: components/org.wso2.carbon.status.dashboard.core/src/main/java/org/wso2/carbon/status/dashboard/core/dbhandler/StatusDashboardMetricsDBHandler.java (hunk 8 above)
Comment (Yasara123) [reply components/org.wso2.carbon.status.dashboard.core/src/main/java/org/wso2/carbon/status/dashboard/core/dbhandler/StatusDashboardMetricsDBHandler.java]: Fixed in d484a92d76bf096f387b03c2bc5b7945dead238b

File: components/org.wso2.carbon.status.dashboard.web/src/gadgets/chart-components/Legend/LegendItem.jsx (hunk 9 above)
Comment (Yasara123) [reply components/org.wso2.carbon.status.dashboard.web/src/gadgets/chart-components/Legend/LegendItem.jsx]: Fixed in bbb066d61f625e5e28b09b91ec2e5a2dc09ed919

File: components/org.wso2.carbon.status.dashboard.web/src/gadgets/chart-components/NumberChart.jsx (hunk 10 above)
Comment (Yasara123) [reply components/org.wso2.carbon.status.dashboard.web/src/gadgets/chart-components/NumberChart.jsx]: Fixed in bbb066d61f625e5e28b09b91ec2e5a2dc09ed919

File: components/org.wso2.carbon.status.dashboard.web/src/gadgets/chart-components/PieChart.jsx (hunk 11 above)
Comment (Yasara123) [reply components/org.wso2.carbon.status.dashboard.web/src/gadgets/chart-components/PieChart.jsx]: Fixed in bbb066d61f625e5e28b09b91ec2e5a2dc09ed919

File: components/org.wso2.carbon.status.dashboard.core/src/main/java/org/wso2/carbon/status/dashboard/core/dbhandler/StatusDashboardMetricsDBHandler.java (hunk 8 above)
Comment (Yasara123) [reply components/org.wso2.carbon.status.dashboard.core/src/main/java/org/wso2/carbon/status/dashboard/core/dbhandler/StatusDashboardMetricsDBHandler.java]: Move to separate repo.

File: components/org.wso2.carbon.status.dashboard.web/src/gadgets/chart-components/Legend/LegendItem.jsx (hunk 9 above)
Comment (Yasara123) [reply components/org.wso2.carbon.status.dashboard.web/src/gadgets/chart-components/Legend/LegendItem.jsx]: Move to separate repo.

File: components/org.wso2.carbon.status.dashboard.web/src/gadgets/chart-components/NumberChart.jsx (hunk 10 above)
Comment (Yasara123) [reply components/org.wso2.carbon.status.dashboard.web/src/gadgets/chart-components/NumberChart.jsx]: Move to separate repo.

File: components/org.wso2.carbon.status.dashboard.web/src/gadgets/chart-components/PieChart.jsx (hunk 12 above)
Comment (Yasara123) [reply components/org.wso2.carbon.status.dashboard.web/src/gadgets/chart-components/PieChart.jsx]: Fixed in bbb066d61f625e5e28b09b91ec2e5a2dc09ed919 .Move to separate repo.

File: components/org.wso2.carbon.status.dashboard.web/src/gadgets/chart-components/PieChart.jsx (hunk 13 above)
Comment (Yasara123) [reply components/org.wso2.carbon.status.dashboard.web/src/gadgets/chart-components/PieChart.jsx]: Fixed in bbb066d61f625e5e28b09b91ec2e5a2dc09ed919 .Move to separate repo.

File: components/org.wso2.carbon.status.dashboard.core/src/main/java/org/wso2/carbon/status/dashboard/core/bean/SiddhiAppStatus.java (hunk 7 above)
Comment (Yasara123) [reply components/org.wso2.carbon.status.dashboard.core/src/main/java/org/wso2/carbon/status/dashboard/core/bean/SiddhiAppStatus.java]: Fixed in 68cfae165beb3c6758c81acd7bcf531a9636dc34

File: components/org.wso2.carbon.siddhi.metrics.core/src/main/java/org/wso2/carbon/siddhi/metrics/core/SiddhiBufferedEventsMetric.java (hunk 4 above)
Comment (Yasara123) [reply components/org.wso2.carbon.siddhi.metrics.core/src/main/java/org/wso2/carbon/siddhi/metrics/core/SiddhiBufferedEventsMetric.java]: Fixed in b4a2bcf444d25031775a679bad76798d70ce7d5b

File: components/org.wso2.carbon.siddhi.metrics.core/src/main/java/org/wso2/carbon/siddhi/metrics/core/SiddhiBufferedEventsMetric.java (hunk 5 above)
Comment (Yasara123) [reply components/org.wso2.carbon.siddhi.metrics.core/src/main/java/org/wso2/carbon/siddhi/metrics/core/SiddhiBufferedEventsMetric.java]: Fixed in b4a2bcf444d25031775a679bad76798d70ce7d5b

File: components/org.wso2.carbon.siddhi.metrics.core/src/main/java/org/wso2/carbon/siddhi/metrics/core/SiddhiLatencyMetric.java (hunk 6 above)
Comment (Yasara123) [reply components/org.wso2.carbon.siddhi.metrics.core/src/main/java/org/wso2/carbon/siddhi/metrics/core/SiddhiLatencyMetric.java]: Fixed in b4a2bcf444d25031775a679bad76798d70ce7d5b

**Pull request merged**